src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -2129,6 +2130,12 @@ protected function childRouteBindingRelationshipName($childType)
      */
     public function resolveRouteBindingQuery($query, $value, $field = null)
     {
+        $columnType = Schema::getColumnType($this->getTable(), $field ?? $this->getRouteKeyName());
```

Comment:
This adds quite a bit of additional SQL overhead to resolving every route binding, as calling `Schema::getColumnType` executes a `select` statement to resolve the column's type:

https://github.com/laravel/framework/blob/0c574d42e9c7d82a0bd304a5cb2ce031a914bfa6/src/Illuminate/Database/Schema/Builder.php#L311-L324

https://github.com/laravel/framework/blob/0c574d42e9c7d82a0bd304a5cb2ce031a914bfa6/src/Illuminate/Database/Schema/Grammars/MySqlGrammar.php#L149-L167

This should be overridden at the application level, not framework level.

Reply:
I think it should not overridden in the application because there is no way to get the column type in the route also avoiding any invalid data from the user.

So, do you have any other method to get the column type without triggering any additional query?

Reply:
No, but it should be opt-in per model that requires it, not globally on all models.

It's an unnecessary performance hit when the 99% use-case is route model binding by ID or slug.